scalafix-core/src/main/scala-3/scalafix/util/TokenList.scala

```diff
@@ -0,0 +1,77 @@
+package scalafix.util
+
+import scala.meta.tokens.Token
+import scala.meta.tokens.Tokens
+import scala.meta.classifiers.XtensionClassifiable
+
+import scalafix.util.Compat._
```

Comment:
the idea of Compat classes is that
- they have different implementations depending on the version
- they are referenced in code common to all versions

In that case, can't you try to keep `TokenList` generic? I see the original one was moved to `scala-2` without changes - instead it should remain in `scala` with minor modifications.

Reply:
my bad, looking at it again the problem was that explicitly defining types `View[T]` in the below methods prevented the `Compat` classes to do their work:

```
  def leadingSpaces(token: Token): View[T] =
    leading(token).takeWhile(_.is[Token.Space])

  def trailingSpaces(token: Token): View[T] =
    trailing(token).takeWhile(_.is[Token.Space])

```

by removing the types it now compiles for the different scala versions